front/app/containers/IdeasNewPage/IdeasNewSurveyForm/index.tsx

```diff
@@ -0,0 +1,282 @@
+        undefined
+      );
+    },
+    [uiSchema]
+  );
+
+  if (!participationMethodConfig || !phaseId) {
+    return null;
+  }
+
+  const uuidRegex = /^[A-F\d]{8}-[A-F\d]{4}-4[A-F\d]{3}-[89AB][A-F\d]{3}-[A-F\d]{12}$/i;
+  const fieldsContainFiles = (formValues) => {
+    for (const key in formValues) {
+      if (uuidRegex.test(formValues[key])) {
+        return true;
+      }
+    };
+    return false;
+  }
+
+  const combineFileValues = (formValues, files) => {
+    for (const key in formValues) {
+      if (uuidRegex.test(formValues[key])) {
+        files.forEach((file) => {
+          if (file.id === formValues[key]) {
+            formValues[key] = {
+              id: file.id,
+              name: file.attributes.name,
+            }
+          }
+        })
+      }
+    };
+    return formValues;
+  }
+
+  // Try and load in a draft idea if one exists
+  useEffect(() => {
+    if (draftIdeaStatus === 'success' && !isNilOrError(draftIdea) && !ideaId && schema) {
+      const formValues = getFormValues(
+        draftIdea,
+        schema
+      );
+
+      if (fieldsContainFiles(formValues)) {
+        console.log('Fields contain files');
+        if (remoteFilesStatus !== 'loading') {
+          console.log(remoteFiles);
+          const combinedFileValues = combineFileValues(formValues, remoteFiles?.data);
+          console.log(combinedFileValues);
+          setInitialFormData(combinedFileValues);
+          setIdeaId(draftIdea.data.id);
+          setLoadingDraftIdea(false);
+        }
+      } else {
+        setInitialFormData(formValues);
+        setIdeaId(draftIdea.data.id);
+        setLoadingDraftIdea(false);
+      }
+
+    } else if (draftIdeaStatus === 'error') {
+      setLoadingDraftIdea(false);
+    }
+  }, [draftIdeaStatus, draftIdea, remoteFilesStatus, remoteFiles, schema]);
+
+  const handleDraftIdeas = async (data: FormValues) => {
+    if (data.publication_status === 'draft') {
+      if (isNilOrError(authUser)) {
+        // Anonymous surveys should not save drafts
+        return;
+      }
+
+      return onSubmit(data, false);
+    } else {
+      return onSubmit(data, true);
+    }
+  }
+
+  const convertFileAttributeIds = (data: FormValues) => {
+    for(const key in data) {
+      if (data[key] && data[key].id && data[key].name) {
+        data[key] = data[key].id;
+      }
+    }
+    console.log(data);
+    return data;
+  }
+
+  // TODO: JS - Still need to get the ID of the file upload into the form data
+  const onSubmit = async (data: FormValues, published?: boolean) => {
+    const postData = convertFileAttributeIds(data);
+    const requestBody = {
+      ...postData,
+      project_id: project.data.id,
+      publication_status: data.publication_status || 'published',
+    };
+
+    // Update or add the idea depending on if we have an existing draft idea
+    const idea = ideaId ? await updateIdea({ id: ideaId, requestBody }) : await addIdea(requestBody);
+    setIdeaId(idea.data.id);
```

Comment:
@EdwinKato need to get the id of the file from the custom fields.

response will return `field_key='UUID'`
we need to insert this uuid into what we have which is `field_key={name: 'name', content: 'base64', id: NEW ID}